networkx/algorithms/community/label_propagation.py

```diff
@@ -0,0 +1,149 @@
+#-*- coding: utf-8 -*-
+#    Copyright (C) 2015 
+#    All rights reserved.
+#    BSD license.
+
+
+"""
+Based on the code donated by Tyler Rush 
+(https://github.com/networkx/networkx/issues/617)
+
+LPA community detection algorithm (based on 'Community Detection via 
+Semi-Synchronous Label Propagation Algorithms' Cordasco and Gargano, 2011
+
+"""
+
+from networkx.utils.decorators import not_implemented_for
+from collections import Counter
+from itertools import groupby
+
+__all__ = ["label_propagation_communities"]
+
+@not_implemented_for('directed')
+def label_propagation_communities(G):
+    """Finds communities in graph using the label propagation method[1]_. This
+    method uses the diffusion of information in the network to identify
+    communities. Not implemented for directed graphs.
+    
+    Parameters
+    ----------
+    G : graph
+    An undirected NetworkX graph.      
+    
+    Returns
+    -------
+    communities : list
+    List of the communities (sets) 
+    
+  
+    References
+    ----------
+    .. [1] Cordasco, G., & Gargano, L. (2010, December). Community detection 
+       via semi-synchronous label propagation algorithms. In Business 
+       Applications of Social Network Analysis (BASNA), 2010 IEEE International 
+       Workshop on (pp. 1-8). IEEE.
+    """
+    coloring = _color_network(G)
+    labeling = _uniquely_label(G)
+    nrounds = 1
+    while not _labeling_complete(labeling, G):
+        nrounds += 1
+        _update_labels(labeling, coloring, G)
+        
+    for label in set(labeling.values()):
```

Comment:
Simpler, something like: `return (set(v) for k, v in groupby(sorted(G, key=labeling.get), key=labeling.get))`.